Source/AtomicEditor/Editors/JSResourceEditor.cpp

```diff
@@ -107,33 +110,44 @@ String getNormalizedPath(const String& path)
     const String& RESOURCES_MARKER = "resources/";
     return path.SubstringUTF8(path.ToLower().Find(RESOURCES_MARKER));
 }
-    
+
 void JSResourceEditor::HandleRenameResourceNotification(StringHash eventType, VariantMap& eventData)
 {
     using namespace RenameResourceNotification;
     const String& newPath = eventData[P_NEWRESOURCEPATH].GetString();
     const String& path = eventData[P_RESOURCEPATH].GetString();
-    
+
     webClient_->ExecuteJavaScript(ToString("HOST_resourceRenamed(\"%s\",\"%s\");", getNormalizedPath(path).CString(), getNormalizedPath(newPath).CString()));
-    
+
     if (fullpath_.Compare(path) == 0) {
         fullpath_ = newPath;
         SetModified(modified_);
     }
 }
-    
+
 void JSResourceEditor::HandleDeleteResourceNotification(StringHash eventType, VariantMap& eventData)
 {
     using namespace DeleteResourceNotification;
     const String& path = eventData[P_RESOURCEPATH].GetString();
-    
+
     webClient_->ExecuteJavaScript(ToString("HOST_resourceDeleted(\"%s\");", getNormalizedPath(path).CString()));
 }
-    
+
 void JSResourceEditor::HandleProjectUnloadedNotification(StringHash eventType, VariantMap& eventData)
 {
     webClient_->ExecuteJavaScript("HOST_projectUnloaded();");
 }
+
+void JSResourceEditor::HandleUserPreferencesChangedNotification(StringHash eventType, VariantMap& eventData)
+{
+    ToolSystem* tsys = GetSubsystem<ToolSystem>();
+    Project* proj = tsys->GetProject();
+    FileSystem* fileSystem = GetSubsystem<FileSystem>();
+    if (fileSystem->FileExists(proj->GetUserPrefsFullPath()))
+    {
+        webClient_->ExecuteJavaScript(ToString("HOST_loadPreferences(\"atomic://%s\");", proj->GetUserPrefsFullPath().CString()));
+    }
+}
```

Comment:
I don't see anything here that couldn't be accomplished in script.  Do you have thoughts on having this event defined/handled native side?